.github/workflows/ci.yml

```diff
@@ -1,70 +1,173 @@
-      - name: Load cache
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Load dependency cache
         uses: actions/cache@v4
         with:
           path: |
             **/node_modules
-          key: ${{ runner.os }}-test-modules-v1-${{ hashFiles('**/yarn.lock') }}
-      - name: Install dependencies
+          key: ${{ runner.os }}-modules-${{ hashFiles('yarn.lock') }}
+      - name: Install dependencies and build
         run: yarn install --frozen-lockfile --ignore-engines
-      - name: Build project
-        run: yarn run build
-      - name: Build project -- esModuleInterop true
-        run: yarn run build:ts --esModuleInterop true
-      - name: Run unit tests (ubuntu, mac)
-        if: ${{ (matrix.os == 'ubuntu-latest') || (matrix.os == 'macos-latest') }}
-        run: export NODE_OPTIONS="--max_old_space_size=4096" && yarn run test-ci:unit
-      - name: Run unit tests (windows)
-        if: ${{ matrix.os == 'windows-latest' }}
-        run: set NODE_OPTIONS=--max-old-space-size=4096 && yarn run test-ci:unit
       - name: Run system tests
-        if: ${{ matrix.os == 'ubuntu-latest' }}
         uses: nick-fields/retry@v3
         with:
           timeout_minutes: 2
           max_attempts: 3
-          command: export NODE_OPTIONS="--max_old_space_size=4096" && yarn run test-ci:system
+          command: yarn run test-ci:system
+        env:
+          NODE_OPTIONS: --max_old_space_size=4096
+
+  browsertest:
+    name: Browser tests
+    strategy:
+      fail-fast: false
+      matrix:
+        browser:
+          - Chrome
+          - Firefox
+          # TODO: uncomment after the Webkit stuff has been sorted out in Karma
+          # - Webkit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Ensure line endings are consistent
+        run: git config --global core.autocrlf input
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Load dependency cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-modules-${{ hashFiles('yarn.lock') }}
+      - name: Install dependencies and build
+        run: yarn install --frozen-lockfile
+      - name: Install ${{ matrix.browser }} via Playwright
+        if: startsWith(matrix.browser, 'Webkit')
+        # The @L will convert 'Webkit' into lowercase 'webkit' which is the name Playwright expects.
+        run: yarn playwright install --with-deps ${BROWSER_NAME@L}
+        env:
+          BROWSER_NAME: ${{ matrix.browser }}
       - name: Run browser tests
-        if: ${{ matrix.os == 'ubuntu-latest' }}
         uses: nick-fields/retry@v3
         with:
-          timeout_minutes: 3
```

Comment:
Why the reduction from 3 to 2 minutes?

Reply:
The main Comunica CI also has the browser test timeout set to 2 minutes, but maybe it needs to be 3 for link traversal. The browser tests do pass with 2 minute timeout, as well, though.

Reply:
Let's keep the 3 minutes. Pretty sure I've set it deliberately at that value.